src/Catalog/Icons/CatalogLeafDataProcessor.cs

```diff
@@ -170,17 +173,40 @@ private async Task ProcessEmbeddedIconAsync(IStorage destinationStorage, Catalog
             var packageUri = _packageStorage.ResolveUri(packageFilename);
             var packageBlobReference = await _packageStorage.GetCloudBlockBlobReferenceAsync(packageUri);
             using (_telemetryService.TrackEmbeddedIconProcessingDuration(item.PackageIdentity.Id, item.PackageIdentity.Version.ToNormalizedString()))
-            using (var packageStream = await packageBlobReference.GetStreamAsync(cancellationToken))
             {
-                var targetStoragePath = GetTargetStorageIconPath(item);
-                var resultUrl = await _iconProcessor.CopyEmbeddedIconFromPackageAsync(
-                    packageStream,
-                    iconFile,
-                    destinationStorage,
-                    targetStoragePath,
-                    cancellationToken,
-                    item.PackageIdentity.Id,
-                    item.PackageIdentity.Version.ToNormalizedString());
+                Stream packageStream;
+                try
+                {
+                    packageStream = await packageBlobReference.GetStreamAsync(cancellationToken);
+                }
+                catch (StorageException ex) when (ex.RequestInformation.HttpStatusCode == 404)
```

Comment:
Nit: consider using `HttpStatusCode.NotFound` instead of `404`. Just a thought, I don't feel strongly about this.